src/components/rangeslider/defaults.js

```diff
@@ -38,6 +44,31 @@ module.exports = function handleDefaults(layoutIn, layoutOut, axName) {
     coerce('autorange', !axOut.isValidRange(containerIn.range));
     coerce('range');
 
+    var subplots = layoutOut._subplots;
+    if(subplots) {
+        var yIds = subplots.yaxis;
+        var yNames = Lib.simpleMap(yIds, axisIds.id2name);
```

Comment:
This is a bit tricky actually - we don't want *all* y axes, we only want the ones that apply to this x axis. It *might* be sufficient to look at `subplots.cartesian`, filtering on the x axis ID and taking the y axis IDs (note that you can't do this with `subplotId.indexOf(xId) !== -1` because `x` is a valid ID - you have to do something like `subplotId.substr(0, subplotId.indexOf('y')) === xId`, like [here](https://github.com/plotly/plotly.js/blob/113b3618732f28d1a2158771544bf4e60e00c084/src/plots/grid.js#L353))

I say *might* because I'm not sure what happens with rangesliders now if we have overlaying x axes... that would be very weird and confusing, so I think I'd be OK ignoring the y data for these axes, but we should confirm that.

Reply:
Couldn't we just use : `layoutOut[yName].anchor === axisIds.name2id(axName)` in the `for` loop ?

Reply:
In principle you can have a y axis shared across several x axes, or positioned as a free axis, and not anchored to the one with the range slider, so unfortunately I don't think that will work.